core/src/main/groovy/noe/ews/workspace/WorkspaceHttpdEws.groovy

```diff
@@ -0,0 +1,231 @@
+            if (!platform.isSolaris()) {
+              JBFile.makeAccessible(new File(tmpHttpd.basedir + "/" + tmpHttpd.binPath + "/httpd"))
+            }
+          } else {
+            // httpd2.4 in JWS3.0
+            JBFile.makeAccessible(new File(tmpHttpd.basedir + "/" + tmpHttpd.binPath + "/httpd"))
+            // Logdir is a symlink to /var/log/httpd
+            tmpHttpd.logDirs.each { String logdir ->
+              JBFile.makeAccessible(new File(tmpHttpd.basedir + "/" + logdir))
+            }
+          }
+        }
+      }
+
+      if (!skipEwsPostinstall) {
+        installEws()
+      }
+    }
+
+    Httpd httpd = serverController.getServerById(serverController.getHttpdServerId()) as Httpd
+    httpd.setHost(hostIpAddress)
+    httpd.updateConfSetBindAddress(hostIpAddress)
+
+    originalTomcatHosts.putAll(originalTomcatHostsIpAddresses())
+
+    serverController.backup()
+
+    log.info('Creating of new WorkspaceHttpdTomcats has finished.')
+  }
+
+  def destroy() {
+    log.info('Destroying of WorkspaceHttpdTomcats has started.')
+    if (platform.isWindows()) {
+      serverController.uninstallWindowsServices()
+    }
+
+    String httpdServerId = serverController.getHttpdServerId()
+    serverController.getServerById(httpdServerId).setHost(hostIpAddress)
+
+    serverController.getTomcatServerIds([TomcatProperties.TOMCAT_MAJOR_VERSION]).each { String tomcatId ->
+      serverController.getServerById(tomcatId).setHost(originalTomcatHosts.get(tomcatId) as String)
+    }
+
+    super.destroy()
+
+    workspaceHttpd?.destroy()
+    workspaceTomcat?.destroy()
+    workspaceApacheDS?.destroy()
+
+    // TODO HP And no basedir was set??
+    //      What about Testing standalone tomcats (this.ews should be sufficient)
+    if (this.ews && !this.rpm && !this.skipInstall) {
+      if (deleteWorkspace) {
+        Cleaner.cleanDirectoryBasedOnRegex(new File(getBasedir()), /.*(jws|jboss-ews|jbcs).*/)
+        log.info('EWS workspace deleted: ' + basedir)
+      }
+      log.info('EWS workspace NOT deleted: ' + basedir)
+    }
+
+    workspaceHttpd = null
+    workspaceTomcat = null
+    workspaceApacheDS = null
+  }
+
+  /**
+   * Initialize the workspace.
+   */
+  void initWorkspaceEws() {
+    log.info('WorkspaceEws.initWorkspaceEws(): BEGIN')
+
+    // Are the paths valid?
+    // TODO validate tomcat paths (nodenames)
+    def validPaths = Library.validatePaths([
+        basedir,
+        workspaceHttpd.basedirHttpd
+    ])
+    if (!validPaths.isEmpty()) {
+      // TODO LP How to better handle this?
+      throw new RuntimeException('Invalid paths' + validPaths.toString())
+    }
+
+    // TODO validate ews installation
+
+    log.info('WorkspaceEws.initWorkspaceEws(): END')
+  }
+
+  /**
+   * Unzips Hibernate Library archive in src/main/resources/hibernate/library
+   * Downloads JDBC drivers
+   */
+  void prepareHibernate(Boolean unzipHibernate = true) {
+    ewsW = new EwsUtils()
```

Comment:
What is the reasoning for creating EwsUtils in multiple methods(multiple times, not checking if it exists already) rather than in the constructor?